### PR TITLE
Ladybird: Go to homepage on init

### PR DESCRIPTION
**Pull request by martinfalisse**
If a setting is checked (true by default), then when starting up Ladybird go to the homepage.

### CODE REVIEW

File: Ladybird/Utilities.cpp [hunk 1]
```diff
@@ -19,6 +19,11 @@ AK::DeprecatedString ak_deprecated_string_from_qstring(QString const& qstring)
     return AK::DeprecatedString(qstring.toUtf8().data());
 }
 
+ErrorOr<AK::String> ak_string_from_qstring(QString const& qstring)
```
Comment (linusg) [inline Ladybird/Utilities.cpp:22]: Why do we have to specify `AK::` for `String` but not `ErrorOr`?

File: Ladybird/Settings.cpp [hunk 2]
```diff
@@ -23,4 +23,14 @@ void Settings::set_homepage(QString const& homepage)
...
 
+bool Settings::go_to_homepage_on_init()
+{
+    return m_qsettings->value("go_to_homepage_on_init", true).toInt() == 1;
+}
+
+void Settings::set_go_to_homepage_on_init(bool const& go_to_homepage_on_init)
+{
+    m_qsettings->setValue("go_to_homepage_on_init", int(go_to_homepage_on_init));
+}
```
Comment (linusg) [inline Ladybird/Settings.cpp:34]: I'm not sure this makes sense. Why would you specify a homepage and then instruct the browser to not load it?

File: Ladybird/main.cpp [hunk 3]
```diff
@@ -110,8 +110,13 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
     window.resize(800, 600);
     window.show();
 
-    if (auto url = get_formatted_url(raw_url); url.is_valid())
+    if (auto url = get_formatted_url(raw_url); url.is_valid()) {
         window.view().load(url);
+    } else {
+        auto maybe_home_url = ak_string_from_qstring(s_settings->homepage());
+        if (!maybe_home_url.is_error())
+            window.view().load(get_formatted_url(maybe_home_url.release_value().bytes_as_string_view()));
```
Comment (linusg) [inline Ladybird/main.cpp:118]: - The OOM error should be propagated via `TRY()`
- What happens if `s_settings->homepage()` is empty?

**Comment by martinfalisse** [comment]
> you haven't answered my second question :)

Oh, if you mean what happens when the input for the homepage setting is set to be empty, then the browser starts at about:home. Feel like I'm missing something..

**Pull request merged**